stablehlo/integrations/python/tests/stablehlo.py

```diff
@@ -227,20 +228,51 @@ def test_minimum_version():
   assert is_semver_format(curr_version)
 
 
-ASM = """
-func.func @test(%arg0: tensor<2xf32>) -> tensor<2xf32> {
-  %0 = stablehlo.add %arg0, %arg0 : (tensor<2xf32>, tensor<2xf32>) -> tensor<2xf32>
-  func.return %0 : tensor<2xf32>
-}
+ASM_FORMAT = """
+func.func @test(%arg0: tensor<{0}>) -> tensor<{0}> {{
+  %0 = stablehlo.add %arg0, %arg0 : (tensor<{0}>, tensor<{0}>) -> tensor<{0}>
+  func.return %0 : tensor<{0}>
+}}
 """
 
+
+@run
+def test_reference_api():
+  # Formatted as (tensor_type, np_value)
+  # Program runs arg + arg, which is used for expected value
+  tests = [
+    # No numpy types for f8
+    ("f16", np.asarray(1, np.float16)),
+    ("f32", np.asarray(2, np.float32)),
+    ("f64", np.asarray(3, np.double)),
+    ("1xi8", np.asarray([4], np.int8)),
+    ("1xi16", np.asarray([5], np.int16)),
+    ("1xi32", np.asarray([-6], np.int32)),
+    # Numpy uint types interpreter as int
```

Comment:
What uint/int types is this comment referring to?

Reply:
The lack of tests for `np.uint8` etc.

Reply:
Gotcha, thanks! I'm having a hard time parsing this comment then. Do you mean numpy uint types are treated as int in the interpreter? If so, IMO it could still be worth having tests for these, even if they are mostly testing the same paths as the signed int tests.

Reply:
I don't see much value, since numpy is an implementation detail of these tests, we'd be testing upstream MLIR handling of DenseElements + numpy.

JAX defines custom lowerings for these types in: https://github.com/google/jax/blob/46913d7e2ed8e6e7839447813d9f04424d9a05b4/jaxlib/hlo_helpers.py#L28

We could add APIs similar to this, at which point I'd agree we should test them, but until then I don't see much value in testing the raw behavior of `DenseElementsAttr.get`

Reply:
Updated the code comments to clarify this